Reduce memory use when generating pages

sasjo opened the pull request:
The `ReportBuilder` kept a list of pages to generate for no obvious reason. This change removes this list and collects and generates pages in a single pass.

sasjo commented:
Updated with changed method signature. The unit tests on internal structure in `ReportBuilderTest` is the reason why the method even needs to exist and have a return value.

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportBuilder.java:
```diff
@@ -149,34 +149,31 @@ private void copyResources(String resourceLocation, String... resources) {
+		return numPages;
+	}
+	
+	private int generatePage(AbstractPage page) {
```
now I think, this method has no sense as this is wrapper for the single method.

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportBuilder.java:
```diff
@@ -149,34 +149,31 @@ private void copyResources(String resourceLocation, String... resources) {
+	
+	private int generatePage(AbstractPage page) {
+		page.generatePage();
+		return 1;
```
also method that returns constant value is part of code smell and should hot happen

sasjo commented:
Restructured code and updated tests to verify using output HTML files instead of return values from private functions. Removed a test that was only testing internal (now removed) code. Code coverage should be unchanged in target file.

damianszczepanik commented:
Thank you for this contribution!
